### ISSUE TITLE
FontSubstitutes fallback

### ISSUE THREAD

**Issue by iamr8** [snowie2000/mactype#576]
I already used ` [FontSubstitutes] ` to change wide system font faces.
but unfortunately, my desired font has no many of characters ( such Arabic unicode support )

I want to know is there any way to add a Fallback font to Profile, as alternative font for some missing unicode characters?

I'm on latest version.

**Comment by snowie2000** [comment]
Duplication of https://github.com/snowie2000/mactype/issues/574#issuecomment-510550681
I'm thinking about the right way to implement it.

Before it happened, you can always add fontlinking in the registry to ask Windows to link to other fonts if the glyph is not available in your font. (and this is the recommended method to do it)

**Comment by sammilucia** [comment]
Three thoughts:
1. Segoe UI would be a good fallback font because it's so complete
2. Adobe Source Pro is also fairly complete to my knowledge (https://github.com/adobe-fonts) and contains serif, sans-serif, and mono fonts
3. Simply have a new Profile setting similar to
```
FallbackFont=<FontName>
```

**Comment by snowie2000** [comment]
I am thinking to insert the original font as the second last option to the fontlink. 

However, there is a big problem: the layout.
Since this is not a system-wide fontlinking, Windows is not aware of the additional fontlink we added, therefore it still calculates the layout of the text based on the original fontlink which should display a default charactor.

**Comment by kailiu42** [comment]
> Before it happened, you can always add fontlinking in the registry to ask Windows to link to other fonts if the glyph is not available in your font. (and this is the recommended method to do it)

Do you mean that if I substitute font A by B, the font link items in the registry for B still applies? I thought it does not. Let me test it.

**Comment by snowie2000** [comment]
If you replace A with B, the fontlink of B applies. Here is the proof:
![QQ截图20190715144824](https://user-images.githubusercontent.com/11767189/61199223-b5536400-a70f-11e9-9130-23d5eebc2952.png)

The upper window has font substitution disabled because it was launched before I enabled the font substitution.

1. Let `幼圆`=`Segoe UI`
2. As you can see from the upper window that the Chinese character "劔" does exist in the font `幼圆` and there is no fontlink set in the registry. (BTW, all the non-symbolic fonts are all implicitly linked to SimSun in Chinese locale eventually).
3. In the lower window, the `Segoe UI` successfully displayed the character "劔" because it is linked to `Microsoft Yahei` in the registry. The `Segoe UI` itself is a pure English font.

**Comment by snowie2000** [comment]
And this is what would happen if you substitute `幼圆` to a font which doesn't have the character with no fontlink or no font in its link list has the character. It is eventually displayed in SimSun.
![QQ截图20190715161747](https://user-images.githubusercontent.com/11767189/61203133-1d0fac00-a71c-11e9-80dc-0af105800097.png)

**Comment by kailiu42** [comment]
To try to fix my issue #574 I've added font link for Roboto, to Segoe UI. However the colon symbol still did not show up. Any idea?

**Comment by snowie2000** [comment]
You can’t link your font back to its original font as it has already been replaced. 

What you did is only linking the font to itself.

**Comment by kailiu42** [comment]
Ah got it. So this can only be supported by MacType.

**Comment by snowie2000** [comment]
Yes, Mactype has to be modified to support it or let windows link to its original font.

**Comment by iamr8** [comment]
And finally, is this feature will be added to the next version of MacType. Or still no way to implement it?

**Comment by snowie2000** [comment]
I have a way to allow Windows to correctly link back to its original font if you set the link for your new font, but it has to be done manually, not automatically because the fontlink in the registry only takes effect after a reboot.

**Comment by sammilucia** [comment]
Sounds manual

What if a user can specify to use certain characters from a font?

E.g.

```
ExcludeCharacters=;:!?*^{}°•~|
```

**Comment by snowie2000** [comment]
No, not even close😕
Users have to manually add fontlink to the registry for their new font (the target font), and they are able to link the target font back to the substituted one.

 E.g.
One replace Tahoma->NewFontA
needs to add
> NewFontA:  Tahoma.ttf

to the registry, so that any glyph missing in font NewFontA can be altered with glyph from tahoma.

However, if you want to make the substitution as perfect as possible or as close as possible as what the  tahoma looks like, you have to fill the registry with the full link from tahoma like this:
>TAHOMA.ttf, Tahoma
> SIMSUN.TTC,SimSun
> MINGLIU.TTC,PMingLiU
> MSGOTHIC.TTC,MS UI Gothic
> BATANG.TTC,Batang

In this way, not only glyphs in tahoma could be used, but glyphs from its fontlink can also be used if you want to display texts in, for example, Chinese, where neither your font nor tahoma are able to.

What I'm trying to do is to make it possible whereas the current version would result in linking the font back to itself, because the font in the fontlink is also being replaced.

It's a tough and inconvenient job to add fontlink manually, I know. But this is the only way I knew that makes Windows link the font for you. And a reboot is necessary.

**Comment by kailiu42** [comment]
> It's a tough and inconvenient job to add fontlink manually, I know. But this is the only way I knew that makes Windows link the font for you. And a reboot is necessary.

This is well acceptable, at least for me, as the registry editing process is just a one time job.

When would I to see a test build with this feature?
